### ISSUE TITLE
computed(expression) doesn't work with asStructure modifier

### ISSUE THREAD

**Issue by yotambarzilay** [mobxjs/mobx#685]
Hey,

I'm using the boxed-computed form, and I'm trying to apply the asStructure modifier:
```javascript
var computed = mobx.computed(mobx.asStructure(() => {
  return {
    prop1: store.a1 || store.a2,
    prop2: store.b
  }
}));
```

As far as I could figure out, this doesn't work because the ```asStructure``` modifier returns an object, while the computed function expects a function as it's parameter in order to create a ```computerExpr```, otherwise it applies the ```computedDecorator``` method.

Not sure if I'm doing something wrong, or if it's simply not supported at the moment.
Would appreciate any input.

**Comment by mweststrate** [comment]
Nice find! This is indeed an overloading issue, as the decorator invoked with one value `@computed(asStructure) get ...` has exactly the same signature as invoking the decorator. Not sure whether there is a quick fix for that. 

A bit more cumbersome approach would be:

```
const computedBox = {
  computed: asStructure(expr)
}
```

Supporting the modifier as seperate param to `mobx.computed` might fix this more nicely (e.g. `computed(expr, mobx.asStructure)`

**Comment by yotambarzilay** [comment]
@mweststrate I took a stab at a fix, allowing the expr in the computed function to either be a function or an IModifierWrapper.
Would be happy if you could take a llok at #689

**Comment by mweststrate** [comment]
Released as 2.6.4. Thanks for the PR!
